core/trino-main/src/test/java/io/trino/execution/TestDropNotNullConstraintConstraintTask.java

```diff
@@ -0,0 +1,172 @@
+import static io.trino.spi.connector.ConnectorCapabilities.NOT_NULL_COLUMN_CONSTRAINT;
+import static io.trino.spi.connector.SaveMode.FAIL;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.sql.QueryUtil.identifier;
+import static io.trino.sql.planner.TestingPlannerContext.plannerContextBuilder;
+import static io.trino.testing.TestingHandles.TEST_CATALOG_NAME;
+import static io.trino.testing.assertions.TrinoExceptionAssert.assertTrinoExceptionThrownBy;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestDropNotNullConstraintConstraintTask
+        extends BaseDataDefinitionTaskTest
+{
+    @Override
+    @BeforeEach
+    public void setUp()
+    {
+        super.setUp();
+        metadata = new MockMetadataWithNotNull(TEST_CATALOG_NAME);
+        plannerContext = plannerContextBuilder().withMetadata(metadata).build();
+    }
+
+    @Test
+    public void testDropNotNullConstraint()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("existing_table");
+
+        metadata.createTable(testSession, TEST_CATALOG_NAME, simpleTable(tableName), FAIL);
+        TableHandle table = metadata.getTableHandle(testSession, tableName).orElseThrow();
+        assertThat(metadata.getTableMetadata(testSession, table).getColumns())
+                .containsExactly(notNullColumn("a", BIGINT), notNullColumn("b", BIGINT));
+
+        getFutureValue(executeDropNotNullConstraint(asQualifiedName(tableName), identifier("b"), false));
+        assertThat(metadata.getTableMetadata(testSession, table).getColumns())
+                .containsExactly(notNullColumn("a", BIGINT), nullableColumn("b", BIGINT));
+    }
+
+    @Test
+    public void testDropNotNullConstraintNotExistingTable()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("not_existing_table");
+
+        assertTrinoExceptionThrownBy(() -> getFutureValue(executeDropNotNullConstraint(asQualifiedName(tableName), identifier("b"), false)))
+                .hasErrorCode(TABLE_NOT_FOUND)
+                .hasMessageContaining("Table '%s' does not exist", tableName);
+    }
+
+    @Test
+    public void testDropNotNullConstraintNotExistingTableIfExists()
+    {
+        QualifiedName tableName = qualifiedName("not_existing_table");
+
+        getFutureValue(executeDropNotNullConstraint(tableName, identifier("b"), true));
+        // no exception
+    }
+
+    @Test
+    public void testDropNotNullConstraintMissingColumn()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("existing_table");
+        metadata.createTable(testSession, TEST_CATALOG_NAME, simpleTable(tableName), FAIL);
+
+        assertTrinoExceptionThrownBy(() -> getFutureValue(executeDropNotNullConstraint(asQualifiedName(tableName), identifier("missing_column"), false)))
+                .hasErrorCode(COLUMN_NOT_FOUND)
+                .hasMessageContaining("Column 'missing_column' does not exist");
+    }
+
```

Comment:
Add tests corresponding to `alter table if exists` 
`io.trino.execution.TestDropNotNullConstraintConstraintTask#executeDropNotNullConstraint`(table, column, true)`

Reply:
I don't understand this suggestion.